src/Sulu/Bundle/EventLogBundle/Infrastructure/Doctrine/Repository/EventRecordRepository.php

```diff
@@ -62,13 +63,16 @@ public function createForDomainEvent(DomainEvent $domainEvent): EventRecordInter
         return $eventRecord;
     }
 
-    public function add(EventRecordInterface $eventRecord): void
+    public function addAndCommit(EventRecordInterface $eventRecord): void
     {
-        $this->entityManager->persist($eventRecord);
-    }
+        // use persister to insert only given entity instead of flushing all managed entities via the entity manager
+        // this prevents flushing unrelated changes and allows to call this method in a postFlush event-listener
+        $this->entityManager->getUnitOfWork()->computeChangeSet(
+            $this->entityManager->getClassMetadata($this->entityRepository->getClassName()),
+            $eventRecord
+        );
 
-    public function commit(): void
-    {
-        $this->entityManager->flush();
+        $this->entityManager->getUnitOfWork()->getEntityPersister(EventRecord::class)->addInsert($eventRecord);
```

Comment:
Would use `$this->entityRepository->getClassName()` instead of `EventRecord::class` here